### ISSUE TITLE
[Bug]: Unable to get NODE environment variables properly in rsbuid.config.mjs file

### ISSUE THREAD

**Issue by xiaochuan1719** [web-infra-dev/rsbuild#429]
### Version

```sh
System:
    OS: Windows 10 10.0.19045
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
    Memory: 7.16 GB / 15.88 GB
  Browsers:
    Edge: Spartan (44.19041.3570.0), Chromium (119.0.2151.44)
    Internet Explorer: 11.0.19041.3570
  npmPackages:
    @rsbuild/core: ^0.0.18 => 0.0.18
    @rsbuild/plugin-image-compress: ^0.0.18 => 0.0.18
    @rsbuild/plugin-vue2: ^0.0.18 => 0.0.18
```

### Details

Unable to get NODE environment variables properly in rsbuid.config.mjs file；But  i can get it in rspack.config.js file when i use rspack directly.

![image](https://github.com/web-infra-dev/rsbuild/assets/38381164/92fe5cc9-4f0f-4893-9cd8-16de98d63f8f)

![image](https://github.com/web-infra-dev/rsbuild/assets/38381164/0cfd6442-1e3e-4eee-a588-723657273f4d)

![image](https://github.com/web-infra-dev/rsbuild/assets/38381164/517e80d1-c0e9-402a-8fb5-a89dd7c5cf99)

### Reproduce link

creating a new project 

### Reproduce Steps

1. `pnpm create rsbuild@latest`  or  `pnpm create rspack@latest`
2. print env variable in rsbuild.config.mjs file

**Comment by chenjiahan** [comment]
Thanks for the feedback! This issue has been fixed in https://github.com/web-infra-dev/rsbuild/pull/425

We will release a new version tomorrow.

**Comment by chenjiahan** [comment]
Released v0.0.19: https://github.com/web-infra-dev/rsbuild/releases/tag/v0.0.19

Please try it 😄
